test/span.js

```diff
@@ -185,31 +184,80 @@ describe('span should', () => {
         assert.isOk(unnormalizedKey in Span._getBaggageHeaderCache());
     });
 
-    describe('with deferred sampling', () => {
-        beforeEach(function () {
-            spanContext._flags = constants.DEFERRED_SAMPLING_MASK;
-        });
-        it('should not pass deferred sampling flag to child spans', () => {
-            let child = tracer.startSpan('child', {childOf: span.context()});
-            assert.notEqual(child.context.flags & constants.DEFERRED_SAMPLING_MASK,
-                            constants.DEFERRED_SAMPLING_MASK);
-            assert.isOk(child._spanContext.samplingFinalized);
+    describe('with upsampling', () => {
+        beforeEach(() => {
+            tracer._upsampling = {enabled: true};
         });
 
-        it('should make a call to the underlying sampler and use the sampling decision when true', () => {
-            let mockSampler = sinon.mock(tracer._sampler);
-            mockSampler.expects('isSampled').withExactArgs('goodOperation', {}).returns(true);
-            let child = tracer.startSpan('goodOperation', {childOf: span.context()});
-            mockSampler.verify();
-            assert.isOk(child.context().isSampled());
-        });
+        function makeTestCase(flags, callSampler, originalSamplingDecision, originalOperation, newSamplingDecision, newOperation, finalSamplingDecision) {
+            return {
+                flags: flags,
+                callSampler: callSampler,
+                name: `Should ${callSampler?"":"not "}call sampler for flag ${flags} ${newOperation?"with modified operationName and samplingDecision":""}`,
+                original: {
+                    operationName: originalOperation,
+                    samplingDecision: originalSamplingDecision,
+                },
+                new: {
+                    operationName: newOperation,
+                    samplingDecision: newSamplingDecision,
+                },
+                finalSamplingDecision: finalSamplingDecision
+            }
+        }
+        const testCases = [
+            makeTestCase(0, true, true, "idli", null, null, true),
```

Comment:
this is quite difficult to read. When we do similar tests in Go we always try to use struct initialization with named fields. Here you don't need the helper method, just define the objects inline with explicit field names. And you can drop false / null assignments, only define fields relevant to the test. 

You also don't have to define an array, you can write the tests like this:

```
describe('upsampling', () =>{
    beforeEach...
    doit('first test description', {
        flags: ...,
        callSampler: ...,
        ...,
    });
    doit(... another test ...)
    // then the actual body
    function doit(name, testCase) {
        it(name, () => {
           // test logic
        });
    }
});
```

The benefit of this structure is that you can easily breakpoint on specific tests and step through, which is much harder when you drive them from an array of data.